TileFormats/VectorData/README.md

```diff
@@ -1,19 +1,213 @@
+The second polygon has 6 indices starting at byte offset `3 * sizeof(uint32)` and ending at `3 * sizeof(uint32) + 6 * sizeof(uint32) - 1`.
+
+**Figure 2**: Example index buffer.
+
+![](figures/indices.jpg)
+
+The number of indices must be a multiple of three. Each consecutive list of three indices is a triangle that must be ordered counter-clockwise. Each index is from the start of the buffer, **NOT** from the offset of the first position of the polygon.
+
+### Positions
+
+The positions buffer is a `uint16` buffer that contains up to three sub-buffers for the polygons, polylines, and points.
+
+The positions are represented by u, v, and height values that are quantized and delta encoded.
+
+| Field | Meaning |
+| --- | --- |
+| u | The horizontal coordinate of the vertex in the tile. When the u value is 0, the vertex is on the western edge of the tile. Then the value is 32767, the vertex is on the eastern edge of the tile. For other values, the vertex's longitude is a linear interpolation between the longitudes of the western and eastern edges of the tile. |
+| v | The vertical coordinate of the vertex in the tile. When the v value is 0, the vertex is on the southern edge of the tile. When the value is 32767, the vertex is on the northern edge of the tile. For other values, the vertex's latitude is a linear interpolation between the latitudes of the southern and northern edges of the tile. |
+| height | The height of the vertex of the tile. When the height value is 0, the vertex's height is equal to `MINIMUM_HEIGHT` from the feature table. When the value is 32767, the vertex's height is equal to `MAXIMUM_HEIGHT` from the feature table. For other values, the vertex's height is a linear interpolation of the minimum and maximum heights. |
+
+The values are then delta and ZigZag encoded. The delta encoding ensures the values are small integers. The ZigZag encoding ensures the values are positive integers. Example encoding code is listed below:
+```javascript
+function zigZag(value) {
+    return ((value << 1) ^ (value >> 15)) & 0xFFFF;
+}
+
+var lastU = 0;
+var lastV = 0;
+var lastHeight = 0;
+
+for (var i = 0; i < length; ++i) {
+    var u = uBuffer[i];
+    var v = vBuffer[i];
+    var height = heightBuffer[i];
+    
+    uBuffer[i] = zigZag(u - lastU);
+    vBuffer[i] = zigZag(v - lastV);
+    heightBuffer = zigZag(height - lastHeight);
+
+    lastU = u;
+    lastV = v;
+    lastHeight = height;
+}
+```
+
+Example decoding code is listed below:
+```javascript
+function zigZagDecode(value) {
+    return (value >> 1) ^ (-(value & 1));
+}
+
+var u = 0;
+var v = 0;
+var height = 0;
+
+for (var i = 0; i < length; ++i) {
+    u += zigZagDecode(uBuffer[i]);
+    v += zigZagDecode(vBuffer[i]);
+    height += zigZagDecode(heightBuffer[i]);
+    
+    uBuffer[i] = u;
+    vBuffer[i] = v;
+    heightBuffer[i] = height;
+}
+```
+
+#### Polygon positions
+
+**Figure 3**: Polygon layout.
+
+![](figures/polygon.jpg)
```

Comment:
Should this diagram also include height values?

Reply:
Polygons don't have height values.The heights are either given by th epolygon minimum and maximum heightsd in the feature table or the global minimum and maximum for the tile. Only points and polylines need ber-vertex heights.